opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1466,6 +1466,43 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChange("cpu") || d.HasChange("vcpu") || d.HasChange("memory") {
+
+		vmState, _, _ := vmInfos.State()
+		vmRequireShutdown := vmState != vm.Poweroff && vmState != vm.Undeployed
```

Comment:
Is it expected to force powerOff the VM if any of `cpu`, `vpcu` or `memory` has changed ?

For the moment the provider does not manage VM states very well.
In your case, if there is an error during this kind of change, you might face an issue for following plans on this VM.
You might need to "taint' the VM to recover it, it can brake the tfstate.

Reply:
@jaypif So you mean to implement a force poweroff if the "normal" poweroff failed to reduce the risk of a broken tfstate?

Reply:
It can be quite long to poweroff a VM and it might fail for lots of reason. SO yes it can help to have this kind of "force".

In addition, you ask to to resize your VM But in the Read function there is no check that it is actually the case. `flattenTemplate` never fill the falue with the one proviedd by OpenNebula. SO you might face a diff in the next plan.

Reply:
@jaypif But a force poweroff

Reply:
Yes I think its a good start for now.